nntrainer/include/nnstreamer_layer.h

```diff
@@ -0,0 +1,115 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2020 Parichay Kapoor <pk.kapoor@samsung.com>
+ *
+ * @file	nnstreamer_layer.h
+ * @date	26 October 2020
+ * @brief	This is class to encapsulate nnstreamer as a layer of Neural Network
+ * @see		https://github.com/nnstreamer/nntrainer
+ * @author	Parichay Kapoor <pk.kapoor@samsung.com>
+ * @bug		No known bugs except for NYI items
+ *
+ */
+
+#ifndef __NNSTREAMER_LAYER_H__
+#define __NNSTREAMER_LAYER_H__
+#ifdef __cplusplus
+
+#include <layer_internal.h>
+#include <nnstreamer-single.h>
+#include <nnstreamer.h>
+#include <tensor.h>
+
+namespace nntrainer {
+
+/**
+ * @class   NNStreamerLayer
+ * @brief   nnstreamer layer
+ */
+class NNStreamerLayer : public Layer {
+public:
+  /**
+   * @brief     Constructor of NNStreamer Layer
+   */
+  NNStreamerLayer(std::string model = "") :
+    Layer(LayerType::LAYER_BACKBONE_NNSTREAMER),
+    modelfile(model),
+    single(nullptr),
+    in_res(nullptr),
+    out_res(nullptr),
+    in_data_cont(nullptr),
+    out_data_cont(nullptr) {
+    trainable = false;
+  }
+
+  /**
+   * @brief     Destructor of NNStreamer Layer
+   */
+  ~NNStreamerLayer();
+
+  /**
+   * @copydoc Layer::forwarding(sharedConstTensors in)
+   */
+  sharedConstTensors forwarding(sharedConstTensors in);
+
+  /**
+   * @copydoc Layer::backwarding(sharedConstTensors in, int iteration)
+   */
+  sharedConstTensors backwarding(sharedConstTensors in, int iteration);
+
```

Comment:
I wonder.. can we do backwarding with nnstreamer?

Reply:
As NNStreamer_layer class extends Layer, where backwarding is pure virtual, this function has to be defined.
So, it defined here and throws not supported if called.